zokrates_core/src/imports.rs

```diff
@@ -173,22 +173,28 @@ impl Importer {
             } else {
                 // to resolve imports, we need a resolver
                 match resolve_option {
-                    Some(resolve) => match resolve(location.clone(), &import.source) {
-                        Ok((mut reader, location, alias)) => {
-                            let mut source = String::new();
-                            reader.read_to_string(&mut source).unwrap();
-
+                    Some(resolve) => match resolve(location.clone(), import.source.to_string()) {
+                        Ok((source, location)) => {
                             let source = arena.alloc(source);
 
-                            let compiled = compile_module(
-                                source,
-                                Some(location),
-                                resolve_option,
-                                modules,
-                                &arena,
-                            )
-                            .map_err(|e| e.with_context(Some(import.source.to_string())))?;
-                            let alias = import.alias.clone().unwrap_or(alias);
+                            // generate an alias from the imported path if none was given explicitely
```

Comment:
This is one of the main changes: in case an alias is not given, we interpret the import target as a Path and use the file name as the symbol name. This introduces the assumption that the import target is a file path, URL would not work here, but we decided against supporting github paths for example so the assumption is valid for now.

Reply:
Previously it was up to the resolver what the default alias should be. Now it's only based in the import target.